typespec/entries/models.tsp

```diff
@@ -50,7 +50,25 @@ model ChatMLImageContentPart<T extends string = string> {
     type: "image_url" = "image_url";
 }
 
-alias ChatMLContentPart<T extends string = string> = ChatMLTextContentPart<T> | ChatMLImageContentPart<T>;
+model ChatMLAnthropicImageSource<T extends string = string> {
+    type: "base64" = "base64";
+    media_type: string;
+    data: T;
+}
+
+model ChatMLAnthropicInnerImageContentPart<T extends string = string> {
+    type: "image" = "image";
+    source: ChatMLAnthropicImageSource<T>;
+}
+
+/** Anthropic image content part */
+model ChatMLAnthropicImageContentPart<T extends string = string> {
+    tool_use_id: string;
```

Comment:
- The `tool_use_id` should be optional to support the case where it's a normal completion request and not responding to a tool call.

- Likewise, the `type` should support other types such as `user`, `assistant`... etc.

Reply:
can you make theses changes and autogenerate @HamadaSalhab ?